From: Yan Jun <yan.jun8@zte.com.cn>
Date: Tue, 23 Aug 2016 12:51:10 +0800
Subject: rgw: fix the wrong return value

Here `parent->readdir` will return negative numbers When there is an error.
so we should just return it.

Signed-off-by: Yan Jun <yan.jun8@zte.com.cn>
(cherry picked from commit c979be7810793534715c24c6a9f7c8b0ed31de8d)

Fixes: http://tracker.ceph.com/issues/17325
Resolves: rhbz#1381693
---
 src/rgw/rgw_file.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/rgw/rgw_file.cc b/src/rgw/rgw_file.cc
index 1f5b35d..e5a7252 100644
--- a/src/rgw/rgw_file.cc
+++ b/src/rgw/rgw_file.cc
@@ -1400,7 +1400,7 @@ int rgw_readdir(struct rgw_fs *rgw_fs,
     return -EINVAL;
   }
   int rc = parent->readdir(rcb, cb_arg, offset, eof, flags);
-  return -rc;
+  return rc;
 }
 
 /*
